src/main/python/deep_qa/layers/noisy_or.py

```diff
@@ -0,0 +1,88 @@
+from keras import backend as K
+from keras.layers import Layer
+from keras.initializations import identity
+from keras.constraints import Constraint, NonNeg
+from keras.regularizers import L1L2Regularizer
+from overrides import overrides
+
+class BetweenZeroAndOne(Constraint):
+    """Constrains the weights to be between zero and one
+    """
+    def __call__(self, p):
+        # handle negatives
+        p *= K.cast(p >= 0., K.floatx())
+        leaky_zeros_mask = K.epsilon() * K.cast(K.equal(p, K.zeros_like(p)), 'float32')
```

Comment:
Can't you just do `K.cast(p == 0., 'float32')`?  Using `K.zeros_like(p)` allocates a new tensor, which shouldn't be necessary here.